var/spack/repos/builtin/packages/mivisionx/package.py

```diff
@@ -34,26 +35,34 @@ def url_for_version(self, version):
 
     variant('build_type', default='Release', values=("Release", "Debug"), description='CMake build type')
 
+    def patch(self):
+        if '@4.2.0' in self.spec:
+            filter_file('${ROCM_PATH}/opencl',
+                        self.spec['rocm-opencl'].prefix,
+                        'amd_openvx/cmake/FindOpenCL.cmake',
+                        string=True)
+
     def flag_handler(self, name, flags):
         spec = self.spec
         protobuf = spec['protobuf'].prefix.include
         if name == 'cxxflags':
             flags.append('-I{0}'.format(protobuf))
         return (flags, None, None)
 
+    depends_on('cmake@3.16.1', type='build')
     depends_on('ffmpeg@4.1.1', type='build')
     depends_on('protobuf@3.5.0', type='build')
-    depends_on('opencv@3.4.6 +calib3d+core+features2d+highgui+imgcodecs+imgproc+video+videoio+videostab+flann', type='build')
+    depends_on('opencv@3.4.6 +calib3d+core+features2d+highgui+imgcodecs+imgproc+video+videoio+flann+photo+objdetect', type='build')
```

Comment:
```suggestion
    depends_on('opencv@:3.4 +calib3d+core+features2d+highgui+imgcodecs+imgproc+video+videoio+flann+photo+objdetect', type='build')
```

Can you maybe update the version constraints of dependencies in this package? It almost never makes sense to fix version numbers of dependency onto their patch version number (x.y.Z), since the Z part usually indicates non-breaking bug fixes and maybe even security updates.